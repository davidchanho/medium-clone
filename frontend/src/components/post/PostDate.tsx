import React from "react";
import { Card } from "react-bootstrap";
import { formatDate } from "../../helpers/formatDate";
import { PostProps } from "../../types";

function PostDate({ post }: PostProps) {
  return (
    <Card.Text className="text-secondary">
      {formatDate(post.date)} {post.readingTime}
    </Card.Text>
  );
}

export default PostDate;
