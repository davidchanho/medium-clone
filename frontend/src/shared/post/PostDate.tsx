import React from "react";
import { Card } from "react-bootstrap";
import { formatDate } from "../../helpers/formatDate";
import { PostProps } from "../../store/posts/types";

function PostDate({ post }: PostProps) {
  const date = formatDate(post.date);
  
  return (
    <Card.Text className="text-secondary">
      {date} &middot; {post.readingTime}
    </Card.Text>
  );
}

export default PostDate;
