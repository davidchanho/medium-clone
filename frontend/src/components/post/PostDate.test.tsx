import React from "react";
import renderer from "react-test-renderer";
import { testPost } from "../../tests";
import PostDate from "./PostDate";

describe("PostDate", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<PostDate {...testPost} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
