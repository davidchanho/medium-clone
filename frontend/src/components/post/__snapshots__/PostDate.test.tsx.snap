// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostDate renders correctly 1`] = `
<p
  className="text-secondary card-text"
>
  Invalid Date
   Â· 
  test
</p>
`;
